File: ItemCount/Module/VB/ItemCount.Module/Controllers/NavigationViewController.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.SystemModule

Public Class NavigationViewController
    Inherits ViewController

    Protected Overrides Sub OnActivated()
        MyBase.OnActivated()
        AddHandler View.ObjectSpace.Committed, AddressOf ObjectSpace_Committed
    End Sub

    Private Sub ObjectSpace_Committed(ByVal sender As Object, ByVal e As System.EventArgs)
        Dim controller As ShowNavigationItemController = Application.MainWindow.GetController(Of ShowNavigationItemController)()

        If controller IsNot Nothing Then
            controller.RecreateNavigationItems()
        End If
    End Sub

    Protected Overrides Sub OnDeactivated()
        RemoveHandler View.ObjectSpace.Committed, AddressOf ObjectSpace_Committed
        MyBase.OnDeactivated()
    End Sub
End Class

File: ItemCount/Module/CS/ItemCount.Module/Controllers/NavigationViewController.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.SystemModule;
using ItemCount.Module.BusinessObjects;

namespace ItemCount.Module.Controllers {
    public class NavigationViewController : ViewController {
        protected override void OnActivated() {
            base.OnActivated();
            View.ObjectSpace.Committed += ObjectSpace_Committed;
        }
        private void ObjectSpace_Committed(object sender, System.EventArgs e) {
            ShowNavigationItemController controller = Application.MainWindow.GetController<ShowNavigationItemController>();
            if(controller != null) {
                controller.RecreateNavigationItems();
            }
        }
        protected override void OnDeactivated() {
            View.ObjectSpace.Committed -= ObjectSpace_Committed;
            base.OnDeactivated();
        }
    }
}
